result.go

```diff
@@ -386,22 +391,40 @@ func (rt resultGrouped) DotResult() []*dot.Result {
 
 // newResultGrouped(f) builds a new resultGrouped from the provided field.
 func newResultGrouped(f reflect.StructField) (resultGrouped, error) {
-	rg := resultGrouped{Group: f.Tag.Get(_groupTag), Type: f.Type}
-
+	g, err := parseGroup(f)
+	if err != nil {
+		return resultGrouped{}, err
+	}
+	rg := resultGrouped{Group: g.name}
```

Comment:
optional: can simplify logic below with:

```suggestion
	rg := resultGrouped{
		Group: g.name,
		Type: f.Type,
		flatten: g.flatten,
	}
```

And then after the validation `switch`:

```
if rg.flatten {
	rg.Type = rg.Type.Elem()
}
```